src/cache/delta_reader.rs

```diff
@@ -16,126 +16,153 @@ pub type SnapshotIterRange<'a> = btree_map::Range<'a, SchemaKey, Operation>;
 /// Read-only data provider that supports a list of snapshots on top of [`DB`].
 /// Maintains total ordering and respects uncommited deletions.
 /// Should not write to underlying [`DB`].
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct DeltaReader {
     /// Set of not commited changes in chronological order.
     /// Meaning that the first snapshot in the vector is the oldest and the latest is the most recent.
     /// If keys are equal, the value from more recent snapshot is taken.
     snapshots: Vec<Arc<SchemaBatch>>,
```

Comment:
Should we wrap `snapshots` in an `Arc` as well? Something like `snapshots: Arc<[Arc<SchemaBatch>]>`. Hard to say how expensive the clone will be without benchmarking though so maybe it's fine.

Reply:
From "application logic" point of view length of this vector is going to be equal to number of non-finalized blocks. 
I don't think it is suppose be more than 50 blocks in the worst case, so I deliberately kept it as regular vector. 

I've added an item to https://github.com/Sovereign-Labs/sovereign-sdk-wip/issues/575 to add benchmarks there

Reply:
Alright!